structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/AbstractAccountAwareResourceService.java

```diff
@@ -0,0 +1,156 @@
+package com.sequenceiq.cloudbreak.structuredevent.service;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.NotFoundException;
+
+import org.hibernate.exception.ConstraintViolationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.dao.DataIntegrityViolationException;
+
+import com.google.common.collect.Sets;
+import com.sequenceiq.cloudbreak.logger.MDCBuilder;
+import com.sequenceiq.cloudbreak.structuredevent.repository.AccountAwareResource;
+import com.sequenceiq.cloudbreak.structuredevent.repository.AccountAwareResourceRepository;
+
+public abstract class AbstractAccountAwareResourceService<T extends AccountAwareResource>
+        implements AccountAwareResourceService<T> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractAccountAwareResourceService.class);
+
+    @Override
+    public T create(T resource, @Nonnull String accountId) {
```

Comment:
maybe `@NotEmpty` would fit better. Should we add this to the other public methods definition also?